services/horizon/internal/sse_test.go

```diff
@@ -0,0 +1,194 @@
+package horizon
+
+import (
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kinecosystem/go/network"
+	"github.com/kinecosystem/go/services/horizon/internal/ingest"
+	"github.com/kinecosystem/go/services/horizon/internal/ledger"
+	"github.com/kinecosystem/go/services/horizon/internal/render/sse"
+	"github.com/kinecosystem/go/services/horizon/internal/test"
+)
+
+// Check subscription gets updates, and sse.Tick() doesnt trigger channel.
+func TestSSEPubsub(t *testing.T) {
+	ht := StartHTTPTest(t, "base")
+	defer ht.Finish()
+
+	ht.App.ticks.Stop()
+
+	subscription := sse.Subscribe("a")
+	defer sse.Unsubscribe(subscription, "a")
+
+	sse.Publish("a")
+
+	select {
+	case <-subscription:
+		// no-op.  Success!
+	case <-time.After(2 * time.Second):
+		t.Error("channel did not trigger")
+		t.FailNow()
+	}
+
+	sse.Tick()
+	select {
+	case <-subscription:
+		t.Error("channel shouldn't trigger after tick")
+		t.FailNow()
+	case <-time.After(2 * time.Second):
+		// no-op. Success!
+	}
+
+}
+
+// Check 2 subscriptions to different topics. Make sure that one topic doesnt
+// raise both channels.
+func TestSSEPubsubMultipleChannels(t *testing.T) {
+	ht := StartHTTPTest(t, "base")
+	defer ht.Finish()
+
+	ht.App.ticks.Stop()
+
+	channelA := sse.Subscribe("a")
+	channelB := sse.Subscribe("b")
+	defer sse.Unsubscribe(channelA, "a")
+	defer sse.Unsubscribe(channelB, "b")
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(channelA chan interface{}, channelB chan interface{}, wg *sync.WaitGroup) {
+		defer wg.Done()
+		select {
+		case <-channelA:
+			select {
+			case <-channelA:
```

Comment:
spaghetti code: instead of doing a select inside a select, just have two select blocks one after the other. i already wrote this comment here somewhere look it up